### PR TITLE
Add canvas to the site for testing crafts

### PR DESCRIPTION
**Pull request by cryck**
FIrst steps in enabling canvas
Add yet another sticker_ids.json to use temporarily
Thanks to @MeckeDev for the code

- [x] Connect selected stickers to canvas
- [x] Load images from our resources
- [x] Use our sticker.json to load data

### CODE REVIEW

File: canvas.html [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+        if (selectedImageIndex !== null && selectedImageIndex < images.length - 1) {
+            const temp = images[selectedImageIndex];
+            images[selectedImageIndex] = images[selectedImageIndex + 1];
+            images[selectedImageIndex + 1] = temp;
+            selectedImageIndex++;
+            drawImages();
+        }
+    }
+
+    document.getElementById('searchInput').addEventListener('input', function(event) {
```
Comment (MeckeDev) [inline canvas.html:228]: If you don't use the Searchbar and dropdown from my Site you can remove this function, it causes just an error if you leave it here

File: canvas.html [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+            ctx.drawImage(image.element, -image.width / 2, -image.height / 2, image.width, image.height);
+            ctx.restore();
+        });
+    }
+
+    function getAdjustedRotation(rotation) {
+        return rotation;
+    }
+
+    function saveImage() {
```
Comment (rammba) [inline canvas.html:76]: This method is unused and totally same as `saveCanvas`

File: canvas.html [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+
+        return idsToFind.map(_id=>stickersById[_id])
+
+    }
+
+    async function placeSticker() {
+        const stickerIds = getStickerIdsFromUrl();
+        const foundStickers = await findStickersById(stickerIds)
+        console.log(foundStickers)
+        const validStickers = foundStickers.filter(sticker => sticker !== null);
```
Comment (rammba) [inline canvas.html:190]: Maybe this can be done in `findStickersById` method?

File: canvas.html [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+            ctx.translate(image.x, image.y);
+            ctx.rotate(getAdjustedRotation(image.rotation));
+            ctx.drawImage(image.element, -image.width / 2, -image.height / 2, image.width, image.height);
+            ctx.restore();
+        });
+    }
+
+    function getAdjustedRotation(rotation) {
+        return rotation;
+    }
```
Comment (rammba) [inline canvas.html:74]: Is this method needed since it doesn't do anything?

File: canvas.html [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+    </div>
+
+</div>
+
+<div id="canvasContainer" class="mt-3">
+    <canvas id="canvas" width="1280" height="800"></canvas>
+</div>
+
+<script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0-alpha1/dist/js/bootstrap.bundle.min.js"></script>
+<script>
```
Comment (rammba) [inline canvas.html:53]: It would be great to put scripts in `.js` files

**Review by rammba — COMMENTED** [review]
I didn't try how this works just put some comments regarding code readabillity.
I'm glad that we're going further with the project, so continue improving things. 😃

File: canvas.html (hunk 3 above)
Comment (cryck) [reply canvas.html]: I would prefer to keep it out of that function for now. I might have another use-case for the function where I don't need to check for validity for example.

File: canvas.html (hunk 5 above)
Comment (cryck) [reply canvas.html]: Agreed on literally everything else, thanks for the review.

File: canvas.html (hunk 3 above)
Comment (rammba) [reply canvas.html]: Sure, I know this is still WIP pull request.

**Comment by cryck** [comment]
@rammba Could you have a quick test when ever you have time today or tomorrow, to ensure I didn't break something?  

https://add-canvas-to-site.sticker-composer.pages.dev/
Just select some stickers and click the add to canvas button above the selected stickers. 

Screenshot 2024-02-15 at 0 58 03

**Comment by rammba** [comment]
I will do it tommorow.

**Comment by rammba** [comment]
I will check it tomorrow.

**Comment by rammba** [comment]
@cryck Canvas looks great for the first version! Good job 😄 

I noticed that canvas won't load stickers if they aren't set in the order in URL.
Eg: sticker on [slot0](https://add-canvas-to-site.sticker-composer.pages.dev/canvas?slot0=sticker-5448) will be shown, but on the [slot1](https://add-canvas-to-site.sticker-composer.pages.dev/canvas?slot1=sticker-5448) won't.

Maybe we can decrease canvas height since it can be unclear which sticker is selected. When there are few stickers grouped, looks like we always select sticker with heighest layer, right?
image

Other than that, everything is fine.

**Comment by cryck** [comment]
@rammba Thank you for testing! Great finds. 
I am thinking, we could display the selected one on the top right on the canvas. The cover sticker is a great idea as well. Will adjust the PR later to do all the changes.

**Pull request merged**